src/backend/utils/misc/queryenvironment.c

```diff
@@ -878,14 +884,23 @@ static bool _ENR_tuple_operation(Relation catalog_rel, HeapTuple tup, ENRTupleOp
 					CacheInvalidateHeapTuple(catalog_rel, newtup, NULL);
 					break;
 				case ENR_OP_UPDATE:
+					/*
```

Comment:
this is a known issue with temp tables/table variables. Why do we want to invalidate cache in the first place? Temp tables/variables are session local and should not impact any one else. So, we should not be even doing this unless we have entries in the catalog and not in ENR.

Reply:
ENR still uses the syscache. We need to invalidate the syscache, otherwise we would keep finding the tuple in the cache even after dropping/altering the table.

Reply:
This also happens if I remove `CacheInvalidateHeapTuple` after creation.

```
[clouddesk3] % sqlcmd -S localhost -U jdbc_user -P 12345678
1> create table #temp(c1 int)
2> insert into #temp values(1)
3> go
Msg 33557097, Level 16, State 1, Server BABELFISH, Line 1
relation "#temp" does not exist
```

Reply:
https://github.com/babelfish-for-postgresql/postgresql_modified_for_babelfish/pull/134, please take a look how we can avoid shared memory invalidation. Also, perf test data is limited as we should have tried 2000 to 4000 connections

Reply:
just for the record, #134 is a follow on for this